healsparse/healSparseMap.py

```diff
@@ -1202,16 +1204,65 @@ def apply_mask(self, mask_map, mask_bits=None, mask_bit_arr=None, in_place=True)
 
     def __getitem__(self, key):
         """
-        Get a single healpix map out of a recarray map, using the default sentinel
-        values.
-        """
-
-        if not self._is_rec_array:
-            raise TypeError("HealSparseMap is not a recarray map")
-
-        return self.get_single(key, sentinel=None)
+        Get part of a healpix map.
+        """
+        if isinstance(key, str):
+            if not self._is_rec_array:
+                raise IndexError("HealSparseMap is not a recarray map, cannot use string index.")
+            return self.get_single(key, sentinel=None)
+        elif isinstance(key, int):
+            # Get a single pixel
+            # Return a single (non-array) value
+            return self.get_values_pix(np.array([key]))[0]
+        elif isinstance(key, slice):
+            # Get a slice of pixels
+            start = key.start if key.start is not None else 0
+            stop = key.stop if key.stop is not None else hp.nside2npix(self._nside_sparse)
+            step = key.step if key.step is not None else 1
+            return self.get_values_pix(np.arange(start, stop, step))
+        elif isinstance(key, np.ndarray):
+            # Make sure that it's integers
+            if not is_integer_value(key[0]):
+                raise IndexError("Numpy array indices must be integers for __getitem__")
+            return self.get_values_pix(key)
+        elif isinstance(key, list):
+            # Make sure that it's integers
+            arr = np.array(key)
+            if not is_integer_value(arr[0]):
+                raise IndexError("List array indices must be integers for __getitem__")
+            return self.get_values_pix(arr)
+        else:
+            raise IndexError("Illegal index type (%s) for __getitem__ in HealSparseMap." %
+                             (key.__class__))
+
+    def __setitem__(self, key, value):
+        """
+        Set part of a healpix map
+        """
+        if isinstance(key, int):
+            # Set a single pixel
+            return self.update_values_pix(np.array([key]), np.array([value]))
+        elif isinstance(key, slice):
+            # Set a slice of pixels
+            start = key.start if key.start is not None else 0
+            stop = key.stop if key.stop is not None else hp.nside2npix(self._nside_sparse)
+            step = key.step if key.step is not None else 1
+            return self.update_values_pix(np.arange(start, stop, step),
+                                          value)
+        elif isinstance(key, np.ndarray):
+            if not is_integer_value(key[0]):
+                raise IndexError("Numpy array indices must be integers for __setitem__")
+            return self.update_values_pix(key, value)
+        elif isinstance(key, list):
+            arr = np.array(key)
+            if not is_integer_value(arr[0]):
```

Comment:
I think any `ndarray` can be also sliced with a list (or `ndarray`) of bools of the same length as the original `ndarray` so that'd be a good addition.